front/lib/actions/mcp_internal_actions/servers/run_agent/conversation.ts

```diff
@@ -160,6 +161,7 @@ export async function getOrCreateConversation(
         // `run_agent` origin will skip adding the conversation to the user history.
         origin: "run_agent",
         selectedMCPServerViewIds: toolsetsToAdd,
+        mainAgentId: mainAgent.sId,
```

Comment:
also need to handle custom_agent -> deep -> task  , mainAgentId should be the first one (custom_agent)